### PR TITLE
Initialize Jupyter Book

### PR DESCRIPTION
**Pull request by weiji14**
Starting with a minimally modified Jupyter Book initialized with `jupyter-book create book/`. Changed the `_config.yml` to use a proper title and the AGU24 logo. Included a Binder launch button and a footer with CC-BY-4.0 license. Also added a JupyterBook badge to the main README.md.

**Preview** at https://www.generic-mapping-tools.org/agu24workshop/intro.html

Jupyter Book page

Steps to run locally:
```bash
cd agu24workshop
mamba env create --file environment.yml
mamba activate agu24workshop

jupyter-book create book/
jupyter-book build book/
```

TODO:
- [x] Initialize basic Jupyter Book (e93ccb90e7383c6919c3cdbc6b23df1eaefc6a6a)
- [x] Setup GitHub Action to build the jupyterbook and deploy to GitHub Pages
  - Refer to https://jupyterbook.org/en/stable/publish/gh-pages.html#automatically-host-your-book-with-github-actions
- [x] Configure https://github.com/GenericMappingTools/agu24workshop/settings/pages so that https://www.generic-mapping-tools.org/agu24workshop/intro.html works
    image
- [ ] Cleanup to remove unneeded files

References:
- https://jupyterbook.org/en/stable/start/your-first-book.html
- https://jupyterbook.org/en/stable/start/publish.html#publish-your-book-online-with-github-pages
- https://github.com/GenericMappingTools/egu22pygmt/pull/3

### CODE REVIEW

File: book/_config.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Book settings
+# Learn more at https://jupyterbook.org/customize/config.html
+
+title: Mastering Geospatial Visualizations with GMT/PyGMT
+author: The PyGMT Team
```
Comment (yvonnefroehlich) [inline book/_config.yml]:
```suggestion
author: The GMT and PyGMT Teams
```

File: book/_toc.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Table of contents
+# Learn more at https://jupyterbook.org/customize/toc.html
+
+format: jb-book
+root: intro
+parts:
+- caption: Details
+  chapters:
+    - title: AGU24 Homepage
+      url: https://www.agu.org/annual-meeting
```
Comment (yvonnefroehlich) [inline book/_toc.yml:10]: At the moment this link seems to resolve the website for the 2024 fall meeting. But I think later we have to update the link, as in 2025 this link will probably resolve the 2025 fall meeting website >.<.

**Review by yvonnefroehlich — APPROVED** [review]

**Review by yvonnefroehlich — COMMENTED** [review]
Thanks for getting started here @weiji14.
So far, I only made a suggestion regarding the authors and a comment regarding the AGU website.

File: book/_toc.yml (hunk 2 above)
Comment (weiji14) [reply book/_toc.yml:10]: Yeah, I wished they just used a permanent URL like https://www.agu.org/fall-meeting-2024 directly. Will try to remember to change this next year!

**Comment by weiji14** [comment]
Thanks @yvonnefroehlich! I'll merge this in first, so that people can start adding things to the table of contents.

**Pull request merged**